soda/core/soda/sodacl/sodacl_parser.py

```diff
@@ -56,6 +56,19 @@
 ]
 
 
+def handle_header_content_is_not_an_object_error(func):
```

Comment:
in the future, we can maybe create an error.py file to store the error handlers

Reply:
What do you think about a name such that's more along the lines of `assert_header_content_is_dict`. It seems to be more close to what the decorator does right?

Another thing with decorators, and again my knowledge is limited and a bit rusty. But I know that when you use the more vanilla approach, what ends up happening is that sometimes the name of the wrapped function (which is the one you actually care about to have in, for example, tracebacks) gets hidden.

However, you can use `functools.wraps` to put all this in order for you. Check this article out maybe: https://realpython.com/primer-on-python-decorators/#who-are-you-really

Reply:
done and thanks for the very useful functools.wraps info :)

Reply:
Awesome, thanks @baturayo. I'll go ahead and approve but make sure @vijaykiran also gives an ok.